cmd/agent/dist/utils/proxy.py

```diff
@@ -49,22 +53,14 @@ def config_proxy_skip(proxies, uri, skip_proxy=False):
     parsed_uri = urlparse(uri)
 
     # disable proxy if necessary
+    # keep keys so `requests` doesn't use env var proxies either
     if skip_proxy:
-        if 'http' in proxies:
-            proxies.pop('http')
-        if 'https' in proxies:
-            proxies.pop('https')
+        proxies['http'] = None
+        proxies['https'] = None
     elif proxies.get('no'):
-        urls = []
-        if isinstance(proxies['no'], basestring):
-            urls = proxies['no'].replace(';', ',').split(",")
-        elif isinstance(proxies['no'], list):
-            urls = proxies['no']
-        for url in urls:
+        for url in proxies['no'].replace(';', ',').split(','):
```

Comment:
@ofek: quick question here: the logic of this method before your changes accounted for the case when `proxies['no']` is a `list` instead of a `basestring`, is there a reason you removed that logic (and can safely assume it's a `basestring`)?

Reply:
I looked through our codebase for uses like that and didn't find any. also, A5 doesn't have that logic